cmd/edit/cluster/cmd.go

```diff
@@ -551,3 +572,103 @@ func parseRFC3339(s string) (time.Time, error) {
 func isExpectedHTTPProxyOrHTTPSProxy(httpProxy, httpsProxy *string, noProxySlice []string, cluster *cmv1.Cluster) bool {
 	return httpProxy == nil && httpsProxy == nil && len(noProxySlice) > 0 && cluster.Proxy() == nil
 }
+
+func auditLogRoleExists(cluster *cmv1.Cluster) bool {
+	return cluster.AWS().AuditLog().RoleArn() != ""
+}
+
+func auditLogEnableOrUpdatePrompt(currentAuditLogRole string) string {
+	if currentAuditLogRole == "" {
+		return "Enable audit log forwarding to AWS CloudWatch"
+	}
+	return fmt.Sprintf("Update existing audit log forwarding role '%s'", currentAuditLogRole)
+}
+
+func setAuditLogForwarding(r *rosa.Runtime, cmd *cobra.Command, cluster *cmv1.Cluster, auditLogArn string) (
+	argValuePtr *string) {
+	if cmd.Flags().Changed("audit-log-arn") {
+		if !aws.IsHostedCP(cluster) {
+			r.Reporter.Errorf("Audit log forwarding to AWS CloudWatch is only supported for Hosted Control Plane clusters")
+			os.Exit(1)
```

Comment:
NP
I think in general it's better to return an error and exit from the main function.